near-contract-standards/src/fungible_token/core.rs

```diff
@@ -100,3 +100,38 @@ pub trait FungibleTokenCore {
     /// Returns the balance of the account. If the account doesn't exist must returns `"0"`.
     fn ft_balance_of(&self, account_id: AccountId) -> U128;
 }
+
+#[cfg(all(not(target_arch = "wasm32"), feature = "unit-testing"))]
+pub mod tests {
+
+    use near_sdk::test_utils::accounts;
+
+    use super::*;
+
+    fn test_transfer_ok(contract: &mut impl FungibleTokenCore) {
+        assert_eq!(contract.ft_balance_of(accounts(2)), U128(1_000_000_000_000_000));
+        assert_eq!(contract.ft_balance_of(accounts(1)), U128(0));
+        let amount = contract.ft_total_supply().0 / 4;
+        contract.ft_transfer(accounts(1), amount.into(), None);
+        assert_eq!(contract.ft_balance_of(accounts(2)).0, contract.ft_total_supply().0 - amount);
+        assert_eq!(contract.ft_balance_of(accounts(1)).0, amount);
+    }
+
+    fn test_transfer_call_ok(contract: &mut impl FungibleTokenCore) {
+        let amount = contract.ft_total_supply().0 / 4;
+        let result = contract.ft_transfer_call(accounts(1), amount.into(), None, "".to_string());
+        match result {
+            PromiseOrValue::Promise(_promise) => {
+                println!("Handle promise here");
```

Comment:
@frol I am having a hard time here trying to get the data from the Promise, I read documentation on how to resolve promises, and they need a callback, the problem is that I dont know if I should make the call back function in here or where I am calling this function.

Reply:
You won't be able to read the content of `_promise`, instead, use [`near_sdk::test_utils::get_created_receipts`](https://docs.rs/near-sdk/latest/near_sdk/test_utils/fn.get_created_receipts.html)